legal-test-fixture/legal_api/api/converter/ExcelWriter.py

```diff
@@ -0,0 +1,280 @@
+            'last_name',
+            'address_type',
+            'street',
+            'street_additional',
+            'city',
+            'region',
+            'country',
+            'postal_code',
+            'delivery_instructions'
+        ]
+        for i, director_address_sheet_heading in enumerate(director_address_sheet_headings):
+            self.__director_address_sheet.write(
+                0, i, format_non_date(director_address_sheet_heading))
+
+        filing_sheet_headings = [
+            'business',
+            'filing_json',
+            'completion_date',
+            'filing_date',
+            'filing_type',
+            'effective_date',
+            'payment_id',
+            'payment_completion_date',
+            'colin_event_id',
+            'status',
+            'paper_only',
+        ]
+        for i, filing_sheet_heading in enumerate(filing_sheet_headings):
+            self.__filing_sheet.write(
+                0, i, format_non_date(filing_sheet_heading))
+
+    def __write_business_to_excel(self, business):
+
+        self.__business_sheet.write(
+            self.__business_sheet_row_index, 0, format_non_date(business.identifier))
+        self.__business_sheet.write(
+            self.__business_sheet_row_index, 1, format_non_date(business.legal_name))
+        self.__business_sheet.write(
+            self.__business_sheet_row_index, 2, format_non_date(business.legal_type))
+        self.__business_sheet.write(
+            self.__business_sheet_row_index, 3, format_date(business.founding_date))
+        self.__business_sheet.write(
+            self.__business_sheet_row_index, 4, format_date(business.dissolution_date))
+        self.__business_sheet.write(
+            self.__business_sheet_row_index, 5, format_date(business.last_ar_date))
+        self.__business_sheet.write(
+            self.__business_sheet_row_index, 6, format_date(business.last_agm_date))
+        self.__business_sheet.write(
+            self.__business_sheet_row_index, 7, format_date(business.fiscal_year_end_date))
+        self.__business_sheet.write(
+            self.__business_sheet_row_index, 8, format_non_date(business.tax_id))
+        self.__business_sheet.write(
+            self.__business_sheet_row_index, 9, format_non_date(business.last_ledger_id))
+        self.__business_sheet.write(self.__business_sheet_row_index, 10, format_non_date(
+            business.last_remote_ledger_id))
+        self.__business_sheet.write(self.__business_sheet_row_index, 11, format_date(
+            business.last_ledger_timestamp))
+        self.__business_sheet.write(
+            self.__business_sheet_row_index, 12, format_date(business.last_modified))
+
+        self.__business_sheet_row_index += 1
+
+        directors = business.directors.all()
+        for director in directors:
+            self.__write_director_to_excel(business.identifier, director)
+
+        delivery_address = business.delivery_address.one_or_none()
```

Comment:
NB: this is changing due to PR #521

Reply:
Resolving on new PR (rebasing is fubared)